File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/Warbot/teams/javascript_level_0/WarBase.js

var Base = JsWarBase.extend({

	action : function() { 
		
		var messageAgents = this.getMessages();

		for (i = 0; i < messageAgents.size(); i++) {
			if(messageAgents.get(i).getMessage() == "whereAreYou") {
				this.setDebugString("I'm here base JS");
				this.sendMessage(messageAgents.get(i).getSenderID(), "here", "");
			}	
		} 
		
		return this.idle(); 
	}

});


var WarBase = new Base();

File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/2017-2018/Warbot/teams/python_test/WarBase.py
from java.util.logging import Logger
logger = Logger.getLogger("WarBase")


st = {}
st['n'] =0

def actionWarBase():
	broadcastMessageToAll("", "")
	actionWarBase.n+=getNbElementsInBag()
	#setDebugString(`st['n']`+`actionWarBase.n`)
	for m in getMessages():
		if m.getMessage() == "whereAreYouBase":
			reply(m,"Here","")
	nbEnv = getNbElementsInBag()
	if nbEnv > 0:
		return createRocketLauncher()

	st['n'] += 1
	requestRole('gr', 'role')
	setDebugString(str(getNumberOfAgentsInRole('gr','role')))
	return idle()

actionWarBase.n = 0
